code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -467,6 +391,29 @@ public void setMessageSetting(final MessageSettings messageSettings) {
         this.settings = messageSettings;
     }
 
+    /**
+     * Recreates the {@link WebView} frame used for displaying the Messaging IAM using the {@link
+     * MessageWebViewRunner}. This method should be called after a device orientation change occurs.
+     *
+     * @param parentViewWidth {@code int} containing the width of the parent activity
+     * @param parentViewHeight {@code int} containing the height of the parent activity
+     */
+    void recreateWebViewFrame(final int parentViewWidth, final int parentViewHeight) {
+        this.parentViewWidth = parentViewWidth;
+        this.parentViewHeight = parentViewHeight;
+
+        try {
+            MessageWebViewRunner messageWebViewRunner = new MessageWebViewRunner(this);
+            messageWebViewRunner.run();
```

Comment:
I believe we can reduce/break one hop of circular dependency of components containing AEPMessage in the following way. The explanation may look long but the change will likely be simple.

`recreateWebViewFrame()` currently is called either from `MessageFragment#onActivityCreated()` and `MessageFragment.layoutChangeListener#onLayoutChange()` - both of which are will be called on the main thread. Even here (rightly), `messageWebViewRunner.run()` is being called directly resulting in the invocation to be on the calling thread (which is always the main thread in this case). 

So, MessageWebViewRunner need not be a runnable any more.  It can just be a utility class with a `show()` with functionality in the run method.  That will allow you would call `messageWebViewRunner.show()` without creating a thread.

Furthermore,  changing this to be a utility class will allow you to pass the `AEPMessage` on demand  to the `show()`  resulting in removing the need to hold a reference to the `AEPMessage` in this class.
The `MessageWebViewRunner` now just has one public method `show(final AEPMessage message)`. 

Now, looking at the constructor, it initializes member variable `settings`, `messageHeight` etc derived from `AEPMessage` itself. I think this can also be done directly just before we call `MessageWebViewRunner#show(AEPMessage)` (formerly run() ).

Given that `MessageWebViewRunner` is stateless we no longer need to create multiple instances of this runner and just use one instance inside AEPMessage on construction. 
Let me know what you think.